rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala

```diff
@@ -164,6 +165,23 @@ object Runtime {
       )
   }
 
+  class BlockTime[F[_]](val timestamp: Ref[F, Par]) {
```

Comment:
Correct me if I'm wrong, but I think both `blockTime` and `shortLeashParams` must not change during a single deploy. I see 3 options for modelling this:

1. Each deploy gets its own Runtime (with parts of it possibly reused between deploys),
2. Runtime stays the same between deploys, so do system processes, but we have a, `DeployContext` (name tentative; all the things that change per deploy but not during it) passed to each `reducer.eval`,
3. We use Ref-s inside the `Runtime` and pray that the `set-s` are done only in the right places.

I'm really worried that we're using the weakest option - and not only for time and shortLeash, but for deploy phlo limit too (which is additionally conflated with deploy cost). What we should be doing probably is 2.

Reply:
I like the idea of a `DeployContext`, although I think the current scheme is safer than perhaps you are giving credit to. Because a deploy is a single `Par` we process it with a single call to `inj`, so in order to change the value of the parameters in the middle of a deploy you would need to actively try to do that by simultaneously running two threads: one for the `inj` call and one for the `set` call.

Regardless, I still think having an immutable `DeployContext` that gets threaded through everything on the Rholang side, and injected once on the Casper side is a better model. Perhaps we should create a ticket for this refactoring as well.

Reply:
Created https://rchain.atlassian.net/browse/RHOL-1003